### ISSUE TITLE
We need better documentation for testing forms with an EntityType field.

### ISSUE THREAD

**Issue by rtfjr86** [symfony/symfony#27368]
Because my repo class extends `ServiceEntityRepository`,
when I use the  `DoctrineTestHelper`
I get the following error:

TypeError: Argument 1 passed to App\Repository\MyRepository::__construct() must implement interface Symfony\Bridge\Doctrine\RegistryInterface, instance of Doctrine\ORM\EntityManager given, called in /Volumes/illuminate/rest-services/instance-management/vendor/doctrine/orm/lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php on line 68

**Comment by curry684** [comment]
That error has absolutely nothing to do with Forms or EntityTypes. I think you are looking for this: https://github.com/symfony/demo/blob/master/src/Repository/UserRepository.php

If you want more help please share some more relevant code.

**Comment by javiereguiluz** [comment]
@rtfjr86 in order to be able to help you, please update your issue description with the information GitHub asked when opening the issue:

```
**Symfony version(s) affected**: x.y.z

**Description**  
<!-- A clear and concise description of the problem. -->

**How to reproduce**  
<!-- Code and/or config needed to reproduce the problem. If it's a complex bug,
     create a "bug reproducer" as explained in:
     https://symfony.com/doc/current/contributing/code/reproducer.html -->

**Possible Solution**  
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->

**Additional context**  
<!-- Optional: any other context about the problem: log messages, screenshots, etc. -->
```

Thanks!

**Comment by rtfjr86** [comment]
I was able to solve my problem. I needed to provide my own repository factory instead of using the [DefaultRepositoryFactory](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php). My factory injects a Registry instead of an EntityManger and Metadata.

`
$config = $this->getOrmConfig();
$this->emMock = DoctrineTestHelper::createTestEntityManager($config);

 private function getOrmConfig()
{
        $config = new Configuration();
        ...
        $config->setRepositoryFactory(new MyRepositoryFactory($this->registryMock));
        return $config;
}
`

Some more context: I have to use a mock entity manager to pre-populate my EntityType field with values before I submit the form for my unit tests.  

It seems to me that Symfony should have better documentation about testing EntityType form fields. Maybe this should be in Doctrine's docs as well.
